### PR TITLE
feat: update KP to KMS and add a new test

### PR DESCRIPTION
**Pull request by maheshwarishikha**
### Description

Update the repository to use the term KMS instead of KP.
It includes changes in README, variable names/description, main module, default example and adds a test for existing KMS.

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Adds support of KMS including Key Protect and HPCS. Key protect term in the code has been renamed to KMS. The variable name `key_protect_instance_id` has been changed to `kms_instance_id` and `key_protect_key_ring_id` changed to `kms_key_ring_id`. Repository name is also changed to `terraform-ibm-kms-key` from `terraform-ibm-key-protect-key`.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,35 @@
-# Key Protect key module
+# KMS key module
 
 [![Stable (With quality checks)](https://img.shields.io/badge/Status-Stable%20(With%20quality%20checks)-green?style=plastic)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 [![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit&logoColor=white)](https://github.com/pre-commit/pre-commit)
 [![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-key-protect-key?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-key/releases/latest)
```
Comment (ocofaigh) [inline README.md:6]: `terraform-ibm-key-protect-key` needs to be updated in this badge url

File: README.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
 
-This module supports creating a standard or root key in an existing key ring and Key Protect instance. You can specify rotation and deletion policies.
+This module supports creating a standard or root key in an existing key ring and KMS instance. You can specify rotation and deletion policies.
```
Comment (ocofaigh) [inline README.md:9]: perhaps explicitly specify it supports Key Protect and HPCS

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see minor comments

File: README.md (hunk 1 above)
Comment (maheshwarishikha) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (maheshwarishikha) [reply README.md]: Made the required changes.

**Review by shemau — APPROVED** [review]
Looks good.

Very picky, I wonder if in README.md

KMS can be Key Protect and Hyper Protect Crypto Services(HPCS) instance.

KMS can be Key Protect or Hyper Protect Crypto Services(HPCS).

File: README.md [hunk 3]
```diff
@@ -1,35 +1,35 @@
...
 [![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit&logoColor=white)](https://github.com/pre-commit/pre-commit)
-[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-key-protect-key?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-key/releases/latest)
+[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-kms-key?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-kms-key/releases/latest)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
 
-This module supports creating a standard or root key in an existing key ring and Key Protect instance. You can specify rotation and deletion policies.
+This module supports creating a standard or root key in an existing key ring and KMS instance. KMS can be Key Protect and Hyper Protect Crypto Services(HPCS) instance. You can specify rotation and deletion policies for the keys.
```
Comment (ocofaigh) [inline README.md:9]: change `and` -> `or` here:

> KMS can be Key Protect **or** Hyper Protect Crypto Services(HPCS) instance

Comment (ocofaigh) [reply README.md:9]: Also are there official doc links we can link to around Key creation (same way we have in `terraform-ibm-key-protect-key-ring` module) ?

Comment (maheshwarishikha) [reply README.md]: updated

**Comment by maheshwarishikha** [comment]
> Looks good.
> 
> Very picky, I wonder if in README.md
> 
> KMS can be Key Protect and Hyper Protect Crypto Services(HPCS) instance.
> 
> KMS can be Key Protect or Hyper Protect Crypto Services(HPCS).

Updated the content. Thanks 👍

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**